kratos/geometries/quadrilateral_3d_4.h

```diff
@@ -784,41 +776,33 @@ template<class TPointType> class Quadrilateral3D4
      * @see DeterminantOfJacobian
      * @see InverseOfJacobian
      */
-    Matrix& Jacobian( Matrix& rResult,
-                              IndexType IntegrationPointIndex,
-                              IntegrationMethod ThisMethod ) const override
+    Matrix& Jacobian( 
+        Matrix& rResult,
+        IndexType IntegrationPointIndex,
+        IntegrationMethod ThisMethod 
+        ) const override
     {
-        //setting up size of jacobian matrix
+        // Setting up size of jacobian matrix
         rResult.resize( 3, 2, false );
-        //derivatives of shape functions
-        ShapeFunctionsGradientsType shape_functions_gradients =
-            CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
-        Matrix ShapeFunctionsGradientInIntegrationPoint =
-            shape_functions_gradients( IntegrationPointIndex );
-        //values of shape functions in integration points
-//         Vector ShapeFunctionsValuesInIntegrationPoint = ZeroVector( 4 );
-//         /*vector<double>*/
-//         ShapeFunctionsValuesInIntegrationPoint = row(
-//                 CalculateShapeFunctionsIntegrationPointsValues( ThisMethod ),
-//                 IntegrationPointIndex );
-
+        // Derivatives of shape functions
+        ShapeFunctionsGradientsType shape_functions_gradients = CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );

```

Comment:
Again....

Reply:
If you get it correctly for your method you don't need to access the index again below.

Reply:
I don't understand